GUI and command updates

Vysares opened the pull request:
- Fixed an issue where the GUI would sometimes appear blurry.
-  Made the GUI more robust against user error and/or brain failure. It now offers more useful info.
- Added a `Close Serial Port` button, so you don't have to stop the exe to reprogram the Teensy.
- Added a command with code 1 that prints a status report of the payload.
- Opening the serial port now automatically triggers a status report by sending command 1.

![Screenshot 2021-03-25 145201](https://user-images.githubusercontent.com/71406782/112542028-b3152a00-8d79-11eb-8f4d-393b48fe4bba.png)

Vysares committed:
Added command to print payload info

Vysares committed:
Added additional protection from user error

Vysares committed:
Update testNS2.py

Vysares committed:
Update commandHandling.cpp

Vysares committed:
Update testNS2.py

Vysares committed:
New build

Vysares committed:
removed garbage files

Vysares committed:
Update testNS2.py

Vysares committed:
Update commandHandling.cpp

Vysares committed:
New build

Vysares committed:
Fixed crash on disconnecting the Teensy

Vysares commented:
Pyserial throws an exception if the Teensy gets disconnected when the port is open. The software will now catch that exception and close the port instead of just crashing. When the Teensy is reconnected, you can just open the port again without restarting the application.

![Screenshot 2021-03-25 152751](https://user-images.githubusercontent.com/71406782/112546174-b9f26b80-8d7e-11eb-89ef-2f86cec6ea8d.png)

Vysares committed:
Update testNS2.py

JashanChopra commented on FSW/src/headers/commandHandling.hpp:
```diff
@@ -18,13 +18,17 @@ bool checkIfCommandAllowed(int command);
+        INFO = 1,                       // Triggers a readback of system data
+
         // Mode Change
-        ENTER_SAFE_MODE = 1,            // enter safemode state
```
Make sure to update the docs if all these command numbers shifted. We ran into that today. Putting in the prevention from sending dangerous commands saved us, so good change there.

Vysares replied:
Will do. I've been holding off since I've just been adding so many commands, but now that we are starting to test with the hardware I'll make sure to keep it up to date

JashanChopra commented on GSW/TestUtility/Source/testNS2.py:
```diff
@@ -56,27 +62,37 @@ def updateLog(text): # writes text to the log
 # read serial #
 def readSerial(): # reads incoming communication from NS2 via the open serial port
-    data = str(teensy.readline())[2:-1]
+    try:
```
try except statements make me so happy

JashanChopra commented on GSW/TestUtility/Source/testNS2.py:
```diff
@@ -98,21 +114,31 @@ def scanPorts(): # finds and lists all available com ports
                 
 # open serial port #
 def openSerialPort(event=None): # opens a new serial connection if port exists
+    if not re.match(r"COM[0-9]+", portField.get()):
```
Was this in response to a specific error, or just checking the regex here for the hell of it?

Vysares replied:
The `openSerialPort` command just looks for any port with a description *containing* the value in the entry field, that means that if you enter something like 'C' in the field, it will try to open every available port.

Vysares replied:
*would try to open every available port. The regex will prevent that now.

Vysares committed:
Fixed bugs and added data streaming from pin

Vysares committed:
Minor changes to timing logic plus some refactoring

Leaving sunrise or sunset mode now saves the buffer even if mode was changed via command.

Vysares committed:
Update timing.cpp

jaretanderson commented on FSW/src/headers/commandHandling.hpp:
```diff
@@ -71,6 +78,7 @@ namespace commandCode {
         EXIT_MAIN_LOOP,                // exits the main loop
 
         // End of list
+        SELF_DESTRUCT,                
```
can't believe I forgot to add this functionality earlier, this will be really nice to have

jaretanderson reviewed (approved):
Looks great, tons of good fixes here
